posthog/management/commands/resync_persons_on_events.py

```diff
@@ -0,0 +1,131 @@
+
+logger = structlog.get_logger(__name__)
+
+
+def run_resync(options):
+    if not options["team_id"]:
+        logger.error("You must specify --team-id to run this script")
+        exit(1)
+
+    TEAM_ID = options["team_id"]
+    QUERY_ID = f"Persons on Events resync team {TEAM_ID}"
+
+    # override to avoid instance lookup in get_parameter function
+    class AsyncMig0006(mig_0006_mod.Migration):
+        def get_parameter(self, name):
+            return self.parameters[name][0]
+
+    mig_0006 = AsyncMig0006(QUERY_ID)
+
+    def run_step(id, sql, settings=None):
+        logger.info(f"Step {id} running {sql}")
+        execute_op_clickhouse(sql, query_id=QUERY_ID, per_shard=True, settings=settings)
+
+    # start
+    mig_0006._clear_temporary_tables(f"resync restart {TEAM_ID}")
+    prep_steps = [
+        # Create temporary tables  - copy/pasta from 0006
+        f"""
+            CREATE TABLE {mig_0006_mod.TEMPORARY_PERSONS_TABLE_NAME} {{on_cluster_clause}} AS {settings.CLICKHOUSE_DATABASE}.person
+            ENGINE = ReplacingMergeTree(version)
+            ORDER BY (team_id, id)
+            SETTINGS index_granularity = 128 {mig_0006_mod.STORAGE_POLICY_SETTING()}
+        """,
+        f"""
+            CREATE TABLE {mig_0006_mod.TEMPORARY_PDI2_TABLE_NAME} {{on_cluster_clause}} AS {settings.CLICKHOUSE_DATABASE}.person_distinct_id2
+            ENGINE = ReplacingMergeTree(version)
+            ORDER BY (team_id, distinct_id)
+            SETTINGS index_granularity = 128
+        """,
+        f"""
+            CREATE TABLE {mig_0006_mod.TEMPORARY_GROUPS_TABLE_NAME} {{on_cluster_clause}} AS {settings.CLICKHOUSE_DATABASE}.groups
+            ENGINE = ReplacingMergeTree(_timestamp)
+            ORDER BY (team_id, group_type_index, group_key)
+            SETTINGS index_granularity = 128 {mig_0006_mod.STORAGE_POLICY_SETTING()}
+        """,
+        # fill them querying only for the selected team_id
+        f"""
+            INSERT INTO {mig_0006_mod.TEMPORARY_PERSONS_TABLE_NAME}(team_id, id, created_at, properties, is_deleted, version)
+            SELECT
+                team_id,
+                id,
+                created_at,
+                properties,
+                0 as is_deleted,
+                ver as version
+            FROM (
+                SELECT
+                    any(team_id) as team_id,
+                    id,
+                    argMax(created_at, version) as created_at,
+                    argMax(properties, version) as properties,
+                    max(version) as ver
+                FROM
+                    person
+                WHERE
+                    person.team_id = {TEAM_ID}
+                GROUP BY
+                    id
+                HAVING
+                    max(is_deleted) = 0
+            )
+        """,
+        f"""
+            INSERT INTO {mig_0006_mod.TEMPORARY_PDI2_TABLE_NAME}(team_id, distinct_id, person_id, is_deleted, version)
```

Comment:
This might time out.

Rather than writing new queries here, let's do _exactly_ what 0006 does. Yes, we'll be copying extra data but the slowest part will still be the mutation.